### PR TITLE
In list-items, better support icons as illustrations, non-link-styled links, empty actionHref, and skeletons

### PR DESCRIPTION
**Pull request by bearfriend**
1. When icons are used as illustrations, match text `color` and remove `border-radius`
2. Allow the default text color to be overridden without affecting hover text color (also, per design docs*, I believe the default _should_ be `ferrite` instead of `celestine`, but I did not make that change here... yet)
3. If `actionHref` is an empty string, treat it as `null`
4. Don't change the `background-color` on hover of skeletons and hide the primary action link (this may be off-base if skeletons are not standardized this way)

\* [Design docs](http://design.d2l/components/lists/#List_Item_Content_Guidelines) say:
> Primary Text - ... This text _can_ styled as a link if the purpose of the list is unclear, or for visual variety in complex list items."

### CODE REVIEW

File: components/list/list-item-mixin.js
Comment (dbatiste) [inline components/list/list-item-mixin.js:124]: We were talking about the `border-radius` amongst the team earlier today, and it sounds like in hindsight we would have preferred the list-item wasn't so opinionated about this property, and that we'd rather let consumers always control their `border-radius`. I think we're just going to run into more cases were it needs to be overridden, and not necessarily be an icon. We'd like to look into the option of perhaps changing this... but obviously existing usages of the component would need to start specifying it if we remove it.

**Comment by dbatiste** [comment]
Oh for...
>Allow the default text color to be overridden without affecting hover text color (also, per design docs*, I believe the default should be ferrite instead of celestine, but I did not make that change here... yet)

I think we'd definitely want to review that with Jeff or whomever "owns" the list design.  I know he has a lot of things in the works for changes to lists.  I think the rationale was that list-item-links should be celestine like standard links, and they also get the underline when hovering/focusing.  The celestine provides the user a hint that they are links before hovering or focusing. I don't necessarily think the design doc is up to date.

File: components/list/list-item-link-mixin.js
Comment (bearfriend) [reply components/list/list-item-link-mixin.js:32]: The skeletons were not covering the link in w2d. I didn't dig into why. We had originally been holding off adding `actionHref` until the skeleton was gone but it was making things more complicated than necessary.

File: components/list/list-item-mixin.js
Comment (bearfriend) [reply components/list/list-item-mixin.js:93]: Oof. I think changing the variable was my intent. This would leak all over the place, wouldn't it?

Not sure I understand about putting it below in the illustration rule. The intent is to update the color of both the primary text and the illustration icon color.

File: components/list/list-item-mixin.js
Comment (bearfriend) [reply components/list/list-item-mixin.js:124]: Right. I was having a hard time thinking of other things people would put in here that _wouldn't_ want the radius, and I figured `d2l-icon` would be a fairly common one. @m6jones had suggested a separate `slot` for icons that maybe could be more generic and not have the radius and that wouldn't disrupt existing usages.

File: components/list/list-item-link-mixin.js
Comment (bearfriend) [reply components/list/list-item-link-mixin.js:19]: Yeah, I ran into this scenario as a bug but figured it was a real possibility so I added it.

File: components/list/list-item-link-mixin.js
Comment (dbatiste) [reply components/list/list-item-link-mixin.js:32]: Ah k.  I'd like to see if I can reproduce this in the demo page. This change might be what we want but I'd like to understand the root cause first.

File: components/list/list-item-mixin.js
Comment (dbatiste) [reply components/list/list-item-mixin.js:93]: By using a variable it means that consumers can override it higher in the DOM, which as far as I know we don't want them to do. I think it's fine to use the variable and we maintain that it is private. I think an alternative is that we also render the `d2l-hovering` and `d2l-focusing` classes on a the illustration slot or a div that wraps it, thereby allowing celestine to be specified explicitly.

File: components/list/list-item-mixin.js
Comment (dbatiste) [reply components/list/list-item-mixin.js:124]: Yeah it's hard to know what they want.  I think for now, we add this rule to unblock you, and we'll start going through the existing usages to apply border radius so we can remove this in the near future without regressions.

File: components/list/list-item-link-mixin.js
Comment (dbatiste) [reply components/list/list-item-link-mixin.js:32]: Ok, I think I see what's happening here.  The list-item layout stacks its action `<a>` on top of the content so that when the user clicks on the item it navigates, or if they right-click then they get all the standard browser actions for links etc. This `<a>` is typically invisible and fills the same dimensions as the content but the user can still interact with it while the skeleton is active.  So... hiding this element I think is the right thing here.

**Review by dbatiste — APPROVED** [review]

**Pull request merged**
